is_core/forms/widgets.py

```diff
@@ -373,3 +373,55 @@ def render(self, name, value, attrs=None):
     def value_from_datadict(self, data, files, name):
         value = super(MultipleTextInput, self).value_from_datadict(data, files, name)
         return [v.strip() for v in value.split(self.separator)] if isinstance(value, six.string_types) else value
+
+
+class AbstractDateRangeWidget(MultiWidget):
+
+    def __init__(self):
+        class_from, class_to = self.get_range_classes()
+        super(AbstractDateRangeWidget, self).__init__(
+            (
+                TextInput(attrs={'class': class_from}),
+                TextInput(attrs={'class': class_to}),
+            )
+        )
+
+    def get_range_classes(self):
+        raise NotImplemented
+
+    def decompress(self, value):
+        return []
+
+
+class DateRangeWidget(AbstractDateRangeWidget):
+
+    def get_range_classes(self):
+        return 'date-range-from', 'date-range-to'
+
+
+class DateTimeRangeWidget(AbstractDateRangeWidget):
+
+    def get_range_classes(self):
+        return 'datetime-range-from', 'datetime-range-to'
+
+
+class FilterDateRangeWidgetMixin(object):
```

Comment:
tady jsem to dost předělal, vytvořil jsem mixin, která upraví widget attrs (všimni si funkce zip) může se ti příště hodit abys nemusel dělat range, obecně range jde v 99.99% obejít tak aby byl kód čitelnější

Reply:
nechtěl jsem aby se data-filter nastavoval v table view. Zaprvé není to univerzální a zadruhé isinstance také ve většině případů poukazuje na kód, který byl špatně navržen (ne vždy u některých obecných řešení být musí např isinstance(s, six.string_types) ale...

Reply:
mixin FilterDateRangeWidgetMixin je dobrej, akorat bych jej podedil primo v DateRangeWidget a DateTimeRangeWidget, nevytvarel bych uz DateRangeFilterWidget a DateTimeRangeFilterWidget

Reply:
nenastavovani data-filter v table view je urcite lepsi reseni, isinstance mas napr pouzite v metode _get_filter_widget v TableViewMixin-u :)

Reply:
já chtěl mít použitelné DateRangeWidget a DateTimeRangeWidget i mimo filtry, mohlo by se to hodit